lightning/src/ln/channelmanager.rs

```diff
@@ -3160,9 +3188,22 @@ where
 				(short_channel_id, amt_to_forward, outgoing_cltv_value, Some(next_packet_pk))
 			},
 			onion_utils::Hop::Forward {
-				next_hop_data: msgs::InboundOnionPayload::BlindedForward { .. }, ..
+				next_hop_data: msgs::InboundOnionPayload::BlindedForward {
+					short_channel_id, ref payment_relay, ref payment_constraints, ref features, ..
+				}, ..
 			} => {
-				todo!()
+				let (amt_to_forward, outgoing_cltv_value) = match Self::check_blinded_forward(
+					msg.amount_msat, msg.cltv_expiry, payment_relay, payment_constraints, features
+				) {
+					Ok(amt) => amt,
```

Comment:
Maybe name this something like `amount_and_cltv`? Or maybe use `map_err()?` instead of `match`?